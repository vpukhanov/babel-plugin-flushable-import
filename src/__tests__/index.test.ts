import plugin from '../index'

import pluginTester from 'babel-plugin-tester'

const babelOptions = {
  babelrc: false,
}

pluginTester({
  plugin,
  pluginName: 'flushable-import',
  babelOptions,
  snapshot: true,
  tests: {
    'static import': 'import("./Foo")',
    'static import (with relative paths)': 'import("../../Foo")',
    'static import (with file extension)': 'import("./Foo.js")',
    'static import (string template)': 'import(`./base`)',
    'static import (string template + relative paths)': 'import(`../../base`)',
    'static import (import as function with relative paths + nested folder)':
      'const obj = {component:()=>import(`../components/nestedComponent`)}; ()=> obj.component()',
    'static import (relative paths + nested folder)':
      'import(`../components/nestedComponent`)',
    'dynamic import (string template)': 'import(`./base/${page}`)',
    'dynamic import (string template with nested folder)':
      'import(`./base/${page}/nested/folder`)',
    'dynamic import (string template with multiple nested folders)':
      'import(`./base/${page}/nested/{$another}folder`)',
    'dynamic import (string template - dynamic at 1st segment)':
      'import(`./${page}`)',
    'dynamic import (string template + relative paths)':
      'import(`../../base/${page}`)',
    'existing chunkName': 'import(/* webpackChunkName: \'Bar\' */"./Foo")',
    'multiple imports': 'import("one"); import("two"); import("three");',
  },
})
