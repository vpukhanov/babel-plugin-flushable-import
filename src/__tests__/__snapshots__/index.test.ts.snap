// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flushable-import dynamic import (string template + relative paths): dynamic import (string template + relative paths) 1`] = `

import(\`../../base/\${page}\`)

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import dynamic import (string template - dynamic at 1st segment): dynamic import (string template - dynamic at 1st segment) 1`] = `

import(\`./\${page}\`)

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import dynamic import (string template with multiple nested folders): dynamic import (string template with multiple nested folders) 1`] = `

import(\`./base/\${page}/nested/{$another}folder\`)

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import dynamic import (string template with nested folder): dynamic import (string template with nested folder) 1`] = `

import(\`./base/\${page}/nested/folder\`)

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import dynamic import (string template): dynamic import (string template) 1`] = `

import(\`./base/\${page}\`)

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import existing chunkName: existing chunkName 1`] = `

import(/* webpackChunkName: 'Bar' */"./Foo")

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import multiple imports: multiple imports 1`] = `

import("one"); import("two"); import("three");

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport3 from "@vpukhanov/babel-plugin-flushable-import/flushable-import";
import _flushableImport2 from "@vpukhanov/babel-plugin-flushable-import/flushable-import";
import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});

_flushableImport2({});

_flushableImport3({});


`;

exports[`flushable-import static import (import as function with relative paths + nested folder): static import (import as function with relative paths + nested folder) 1`] = `

const obj = {component:()=>import(\`../components/nestedComponent\`)}; ()=> obj.component()

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";
const obj = {
  component: () => _flushableImport({}),
};

() => obj.component();


`;

exports[`flushable-import static import (relative paths + nested folder): static import (relative paths + nested folder) 1`] = `

import(\`../components/nestedComponent\`)

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import static import (string template + relative paths): static import (string template + relative paths) 1`] = `

import(\`../../base\`)

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import static import (string template): static import (string template) 1`] = `

import(\`./base\`)

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import static import (with file extension): static import (with file extension) 1`] = `

import("./Foo.js")

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import static import (with relative paths): static import (with relative paths) 1`] = `

import("../../Foo")

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;

exports[`flushable-import static import: static import 1`] = `

import("./Foo")

      ↓ ↓ ↓ ↓ ↓ ↓

import _flushableImport from "@vpukhanov/babel-plugin-flushable-import/flushable-import";

_flushableImport({});


`;
